iceoryx_hoofs/source/posix_wrapper/shared_memory_object.cpp

```diff
@@ -166,17 +166,36 @@ SharedMemoryObject::SharedMemoryObject(SharedMemory&& sharedMemory,
 {
 }
 
-void* SharedMemoryObject::allocate(const uint64_t size, const uint64_t alignment) noexcept
+cxx::expected<void*, SharedMemoryAllocationError> SharedMemoryObject::allocate(const uint64_t size,
+                                                                               const uint64_t alignment) noexcept
 {
-    return m_allocator.allocate(size, alignment);
+    if (size == 0)
+    {
+        IOX_LOG(WARN) << "Cannot allocate memory of size 0.";
+        return cxx::error<SharedMemoryAllocationError>(SharedMemoryAllocationError::REQUESTED_ZERO_SIZED_MEMORY);
+    }
+    if (m_allocationFinalized)
```

Comment:
This error condition in turn is a bit on the fence. Arguably one can request the system designer to ensure to never call it after finalization (if he controls finalization as well). On the other hand it is much harder to do when finalization is controlled elsewhere. So this is a legitimate error case that needs to be communicated somehow.

Same with being out of memory at runtime. Controllable to some degree, but not in all usage scenarios.

Resolve me.